examples/dual_variable_example/dual_variable_example.py

```diff
@@ -0,0 +1,287 @@
+# ****************************************************************************
+
+###############################################################################
+# imports
+###############################################################################
+
+import logging
+import os
+import pprint as pp
+
+import pandas as pd
+from oemof.solph import EnergySystem, Model, buses
+from oemof.solph import components as cmp
+from oemof.solph import flows, processing, views
+from oemof.tools import logger
+
+try:
+    import matplotlib.pyplot as plt
+except ImportError:
+    plt = None
+
+
+solver = "cbc"  # 'glpk', 'gurobi',....
+debug = False  # Set number_of_timesteps to 3 to get a readable lp-file.
+number_of_time_steps = 48
+solver_verbose = False  # show/hide solver output
+
+# initiate the logger (see the API docs for more information)
+logger.define_logging()
+
+date_time_index = pd.date_range(
+    "1/1/2012", periods=number_of_time_steps, freq="H"
+)
+
+energysystem = EnergySystem(timeindex=date_time_index)
+
+demand = [
+    209,
+    207,
+    200,
+    191,
+    185,
+    180,
+    172,
+    170,
+    171,
+    179,
+    189,
+    201,
+    208,
+    207,
+    205,
+    206,
+    217,
+    232,
+    237,
+    232,
+    224,
+    219,
+    223,
+    213,
+    201,
+    192,
+    187,
+    184,
+    184,
+    182,
+    180,
+    191,
+    207,
+    222,
+    231,
+    238,
+    241,
+    237,
+    234,
+    235,
+    242,
+    264,
+    265,
+    260,
+    245,
+    238,
+    241,
+    231,
+]
+pv = [
+    0.18,
+    0.11,
+    0.05,
+    0.05,
+    0.0,
+    0.0,
+    0.0,
+    0.0,
+    0.0,
+    0.0,
+    0.0,
+    0.0,
+    0.0,
+    0.05,
+    0.07,
+    0.11,
+    0.13,
+    0.15,
+    0.22,
+    0.28,
+    0.33,
+    0.25,
+    0.17,
+    0.09,
+    0.09,
+    0.07,
+    0.05,
+    0.05,
+    0.0,
+    0.0,
+    0.0,
+    0.0,
+    0.0,
+    0.0,
+    0.0,
+    0.0,
+    0.0,
+    0.09,
+    0.21,
+    0.33,
+    0.44,
+    0.54,
+    0.61,
+    0.65,
+    0.67,
+    0.64,
+    0.59,
+    0.52,
+]
+
+##########################################################################
+# Create oemof object
+##########################################################################
+
+# create natural gas bus
+bus_gas = buses.Bus(label="natural_gas")
+
+# create electricity bus
+bus_elec = buses.Bus(label="electricity")
+
+# adding the buses to the energy system
+energysystem.add(bus_gas, bus_elec)
+
+# create excess component for the electricity bus to allow overproduction
+energysystem.add(cmp.Sink(label="excess_bel", inputs={bus_elec: flows.Flow()}))
+
+# create source object representing the natural gas commodity (annual limit)
+energysystem.add(
+    cmp.Source(
+        label="rgas",
+        outputs={bus_gas: flows.Flow(variable_costs=38)},
+    )
+)
+
+# create fixed source object representing pv power plants
+energysystem.add(
+    cmp.Source(
+        label="pv",
+        outputs={bus_elec: flows.Flow(fix=pv, nominal_value=700)},
```

Comment:
This is a matter of taste. I prefer to use fix+excess because I will find the excess in the excess variable. But it is also possible to subtract the resulting `PV_out` variable from the given max value to get the excess.

Reply:
This is why I put it as a question. You don't need to change this.